internal/controllers/machinehealthcheck/machinehealthcheck_targets.go

```diff
@@ -155,6 +156,9 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 		if conditions.IsTrue(t.Cluster, clusterv1.InfrastructureReadyCondition) && clusterInfraReady != nil && clusterInfraReady.Time.After(comparisonTime) {
 			comparisonTime = clusterInfraReady.Time
 		}
+		if conditions.IsTrue(t.Machine, clusterv1.InfrastructureReadyCondition) && machineInfraReady != nil && machineInfraReady.Time.After(comparisonTime) {
```

Comment:
It'd be nice to complement tests
"when the node has not yet started for longer than the timeout",
And
"when the node has not yet started for shorter than the timeout",

to account for some of these conditions.

Reply:
Yes once we agree this is what we want, I'll add some tests

Reply:
added